### ISSUE TITLE
对于MSSA.py中的模型部分的一点问题

### ISSUE THREAD

**Issue by 879068** [Feaxure-fresh/TL-Fault-Diagnosis-Library#8]
我目前正在学习您的代码。主要运用了MSSA.py这个模型。但是，其中，在MSSA.py中，在self.G_shared = model_base.FeatureExtractor(in_channel=1).to(self.device)→转到model_base.py的class FeatureExtractor(nn.Module)这里时，window_sizes=[4, 8, 16, 24, 32]应该是卷积核的设置吧，但该代码与原文的卷积核设置应该是不一样的吧。
原文的结构如下：
Layer1: Conv(16*(1*32)) BN(16) MP(1*2)
Layer2: Conv(32*(1*3)) BN(32) MP(1*2)
Layer1: Conv(64*(1*3)) BN(64) MP(1*2)
Layer2: Conv(128*(1*3)) BN(128) MP(1*2)
Layer3: Conv(128*(1*3)) BN(128) MP(1*2)

其次，就是我有一点不理解的地方就，代码运行后，形成的ModuleList，为什么，每一个模型的卷积核都是一样的，为什么会有那么多个模型呢。
例如：
FeatureExtractor(
  (convs): ModuleList(
    (0): CNNlayer(
      (fs): Sequential(
        (0): Sequential(
          (0): Conv1d(1, 4, kernel_size=(4,), stride=(1,), padding=(1,))
          (1): BatchNorm1d(4, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
          (2): ReLU(inplace=True)
          (3): MaxPool1d(kernel_size=2, stride=2, padding=0, dilation=1, ceil_mode=False)
        )
        (1): Sequential(
          (0): Conv1d(4, 16, kernel_size=(4,), stride=(1,), padding=(1,))
          (1): BatchNorm1d(16, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
          (2): ReLU(inplace=True)
          (3): MaxPool1d(kernel_size=2, stride=2, padding=0, dilation=1, ceil_mode=False)
        )
        (2): Sequential(
          (0): Conv1d(16, 32, kernel_size=(4,), stride=(1,), padding=(1,))
          (1): BatchNorm1d(32, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
          (2): ReLU(inplace=True)
          (3): MaxPool1d(kernel_size=2, stride=2, padding=0, dilation=1, ceil_mode=False)
        )
        (3): Sequential(
          (0): Conv1d(32, 64, kernel_size=(4,), stride=(1,), padding=(1,))
          (1): BatchNorm1d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
          (2): ReLU(inplace=True)
          (3): MaxPool1d(kernel_size=2, stride=2, padding=0, dilation=1, ceil_mode=False)
        )
        (4): Dropout(p=0.0, inplace=False)
        (5): Sequential(
          (0): Conv1d(64, 128, kernel_size=(4,), stride=(1,), padding=(1,))
          (1): BatchNorm1d(128, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
          (2): ReLU(inplace=True)
          (3): AdaptiveMaxPool1d(output_size=4)
          (4): Flatten(start_dim=1, end_dim=-1)
        )
      )
    )
这里面的卷积核全为4

打扰了。

**Comment by Feaxure-fresh** [comment]
我们提供的代码主旨不在于精确再现文献中的全部配置，而是旨在相同模型架构条件下，评估不同迁移学习方法的效能。考虑到部分文献会公开其模型架构，而另一些则不会，且这些架构设置常常存在差异，因此，若每个模型采用独特的架构设置，便无法有效地只针对迁移学习方法进行性能比较。此外，使用那些未公开代码的文献方法时，请注意，这些代码是基于我们对原文的理解而复现的，可能与原文实际内容有所差异（有时所提方法的部分内容未在代码中使用）。因此，在测试或应用这些模型前，我们强烈建议您仔细阅读原始文献，并检查相关代码。同时，我们非常欢迎所有研究者分享对模型代码更正的见解。

关于第二个问题，ModuleList中含有相同卷积核的多个CNN层的设计目的，在于通过逐层处理低通道输入，以获得多通道、高复杂度的特征表示。尽管这些CNN层使用相同的卷积核，但它们处理的输入和输出通道数不同，这种做法是深度卷积网络构建中的一种常见策略。

**Comment by 879068** [comment]
感谢您对我的问题的解答。也十分感谢您提供的迁移学习的代码，我收获很多。
